src/concern/mod.rs

```diff
@@ -64,6 +64,23 @@ impl ReadConcern {
     pub fn custom(level: String) -> Self {
         ReadConcernLevel::from_str(level.as_str()).into()
     }
+
+    pub(crate) fn serialize_for_client_options<S>(
```

Comment:
I'm not too excited about doing that, since adding an extra struct in the `ClientOptions` serializer reduces readability. I'd like to keep the `Serialize` logic of the nested structs abstracted away as much as possible.

However, I do agree that in the case of `ReadConcern`, the custom serializer is only doing one thing and my solution is a little overkill.

As a happy medium, how do you feel about directly parsing the level in the `ClientOptionsHelper` struct? Just omit the middleman struct. We can meet up today if you aren't sure what I mean exactly.

Reply:
I implemented the helper in the `ReadConcern` implementation. I think this is a good compromise between readability and compartmentalization, but I'd like another opinion.

Reply:
Given that such methods will only ever be used from within the `ClientOptions` `Serialize` implementation, it does seem fine to me to have them all encapsulated there, and I don't think readability would be too much of an issue. That being said, the current implementation strategy also seems fine, so I don't think it's worth changing them to be consolidated.